src/julia_threads.h

```diff
@@ -124,6 +124,14 @@ typedef struct {
 
 struct _jl_bt_element_t;
 
+typedef struct {
+    uint64_t start_time;
+    uint64_t sleep_time;
+    uint64_t scheduler_time;
+    uint64_t lock_spin_time;
```

Comment:
to do: add `compile_time` (which could be a subset of `lock_spin_time` i guess or we could stop accumulating lock_spin_time when compilation_time starts?)

in future maybe we could split lock_spin_time to have timing for a few important internal locks (like the codegen_lock) but i think that can be follow-up work?